Form/Type/BooleanType.php

```diff
@@ -36,39 +35,26 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
     /**
      * {@inheritdoc}
-     *
-     * @todo Remove it when bumping requirements to SF 2.7+
-     */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
-    {
-        $this->configureOptions($resolver);
-    }
-
-    /**
-     * {@inheritdoc}
      */
     public function configureOptions(OptionsResolver $resolver)
     {
         $choices = array(
-            self::TYPE_YES => 'label_type_yes',
-            self::TYPE_NO => 'label_type_no',
+            'label_type_yes' => self::TYPE_YES,
+            'label_type_no' => self::TYPE_NO,
         );
 
         $defaultOptions = array(
             'transform' => false,
             'translation_domain' => 'SonataCoreBundle',
         );
 
-        // NEXT_MAJOR: Remove this "if" (when requirement of Symfony is >= 2.7)
-        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
-            $choices = array_flip($choices);
+        $choices = array_flip($choices);
```

Comment:
Something is strange here... if you flip the choices every time, wouldn't be better to just flip them in the array creation just above?

Also, if we are always setting them on the `$defaultOptions['choices']`, IMO it is better to just declare them on that array.